codespeed/views.py

```diff
@@ -251,126 +257,189 @@ def gettimelinedata(request):
-                            val_min = res.val_min
-                        val_max = ""
-                        if res.val_max is not None:
-                            val_max = res.val_max
-                        q1 = ""
-                        if res.q1 is not None:
-                            q1 = res.q1
-                        q3 = ""
-                        if res.q3 is not None:
-                            q3 = res.q3
-                        results.append(
-                            [
-                                res.revision.date.strftime('%Y/%m/%d %H:%M:%S %z'),
-                                res.value, val_max, q3, q1, val_min,
-                                res.revision.get_short_commitid(), res.revision.tag, branch.name
-                            ]
-                        )
-                    else:
-                        std_dev = ""
-                        if res.std_dev is not None:
-                            std_dev = res.std_dev
-                        results.append(
-                            [
-                                res.revision.date.strftime('%Y/%m/%d %H:%M:%S %z'),
-                                res.value, std_dev,
-                                res.revision.get_short_commitid(), res.revision.tag, branch.name
-                            ]
-                        )
-                timeline['branches'][branch.name][executable.id] = results
-                append = True
-
-        if baselinerev is not None and append:
-            try:
-                baselinevalue = Result.objects.get(
-                    executable=baselineexe,
-                    benchmark=bench,
-                    revision=baselinerev,
-                    environment=environment
-                ).value
-            except Result.DoesNotExist:
-                timeline['baseline'] = "None"
-            else:
-                # determine start and end revision (x axis)
-                # from longest data series
-                results = []
-                for branch in timeline['branches']:
-                    for exe in timeline['branches'][branch]:
-                        if len(timeline['branches'][branch][exe]) > len(results):
-                            results = timeline['branches'][branch][exe]
-                end = results[0][0]
-                start = results[len(results) - 1][0]
-                timeline['baseline'] = [
-                    [str(start), baselinevalue],
-                    [str(end), baselinevalue]
-                ]
-
-        if append:
-            timeline_list['timelines'].append(timeline)
-
-    if not len(timeline_list['timelines']) and data['ben'] != 'show_none':
-        response = 'No data found for the selected options'
-        timeline_list['error'] = response
-    return HttpResponse(json.dumps(timeline_list))
+        number_of_revs = int(data.get('revs', 10))
+    return number_of_revs, benchmarks
+
+
+def get_stats_with_defaults(res):
```

Comment:
Nice factoring this functions out of the views! They could be moved out of the views module

Reply:
Yeah, indeed, good point. Where would you want them to be? Not sure it fits into the model?

Reply:
`stats` or `view_data` or similar?

Reply:
`view_data` sounded like a good fit. So, I pushed https://github.com/tobami/codespeed/pull/255/commits/503beb5d5f9c23b3d6d1a4e61ba92673017d2597 and moved both helper functions there.